tasks/configure_storage_paths.yml

```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Get stat of the directory {{ __mssql_storage_path }}
+  stat:
+    path: "{{ __mssql_storage_path }}"
+  register: __mssql_storage_path_stat
+
+- name: Ensure the directory and permissions for {{ __mssql_storage_path }}
+  # null converts to an empty string when setting vars in the parent task
+  when: __mssql_storage_mode != ''
+  file:
+    path: "{{ __mssql_storage_path }}"
+    state: directory
+    owner: mssql
+    group: mssql
+    mode: "{{ __mssql_storage_mode }}"
```

Comment:
I think you can do this with a single `file` module
```yaml
    mode: "{{ __mssql_storage_mode is None | ternary(
      __mssql_storage_path_stat is exists | ternary(
       __mssql_storage_path_stat.stat.mode, '0700'),
      __mssql_storage_mode) }}"
```
or if you prefer the more verbose version
```yaml
    mode: "{{ __mssql_storage_mode
      if __mssql_storage_mode is not None else
      __mssql_storage_path_stat.stat.mode if
      __mssql_storage_path_stat is exists else
      '0700' }}"
```

Reply:
or if you want the default mode to be configurable, then add that as a parameter to configure_storage_paths.yml e.g.
```yaml
    mode: "{{ __mssql_storage_mode
      if __mssql_storage_mode is not None else
      __mssql_storage_path_stat.stat.mode if
      __mssql_storage_path_stat is exists else
      __mssql_storage_mode_default }}"
```

Reply:
Oh, that's good, I didn't know that you can do `mode: ""` and the file module will consider this an unset mode, hence I created another task with mode literally not set.

`__mssql_storage_path_stat is exists` doesn't work for me, see below. I used `__mssql_storage_path_stat.stat.exists` instead.

```
TASK [linux-system-roles.mssql : debug] *************************************************************************************************************************************
ok: [127.0.0.3] => {
    "__mssql_storage_path": "/tmp/log"
}

TASK [linux-system-roles.mssql : debug] *************************************************************************************************************************************
ok: [127.0.0.3] => {
    "__mssql_storage_path_stat.stat.path": "/tmp/log"
}

TASK [linux-system-roles.mssql : debug] *************************************************************************************************************************************
ok: [127.0.0.3] => {
    "__mssql_storage_path_stat.stat.path is exists": false
}

TASK [linux-system-roles.mssql : debug] *************************************************************************************************************************************
ok: [127.0.0.3] => {
    "__mssql_storage_path_stat.stat.exists": true
}

TASK [linux-system-roles.mssql : debug] *************************************************************************************************************************************
ok: [127.0.0.3] => {
    "__mssql_storage_path is exists": false

```